test/cqlpy/test_describe.py

```diff
@@ -1197,6 +1197,97 @@ def test_config(altered_cdc_log_table_opt):
     test_config("speculative_retry = '17.0PERCENTILE'")
     test_config("tombstone_gc = {'mode': 'immediate', 'propagation_delay_in_seconds': '17'}")
 
+# Verify that describing the underlying materialized view of a secondary index
+# produces a `CREATE MATERIALIZED VIEW` statement.
+# Reproducer of scylladb/scylladb#24610.
+def test_describe_underlying_mv_of_index(scylla_only, cql, test_keyspace):
+    index_name = "i"
+    mv_name = f"{index_name}_index"
+
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as table:
+        with new_secondary_index(cql, table, "v", index_name):
+            result = cql.execute(f"DESCRIBE MATERIALIZED VIEW {test_keyspace}.{mv_name}").one()
+
+            # Sanity checks.
+            assert result.keyspace_name == test_keyspace
+            assert result.type == "view"
+            assert result.name == mv_name
+
+            assert hasattr(result, "create_statement")
+            assert f"CREATE MATERIALIZED VIEW {test_keyspace}.{mv_name}" in result.create_statement
+
+# Verify that describing the underlying materialized view of an unnamed secondary index
+# produces a `CREATE MATERIALIZED VIEW` statement.
+# Reproducer of scylladb/scylladb#24610.
+def test_describe_underlying_mv_of_unnamed_index(scylla_only, cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "p int PRIMARY KEY, v int") as table:
+        _, table_name = table.split(".")
+        index_name = f"{table_name}_v_idx"
+        mv_name = f"{index_name}_index"
+
+        # We need to create an index on our own because NOT providing a name
+        # in `new_secondary_index` leads to creating an index with a random name,
+        # not an unnamed index.
+        try:
+            cql.execute(f"CREATE INDEX ON {table}(v)")
+            result = cql.execute(f"DESCRIBE MATERIALIZED VIEW {test_keyspace}.{mv_name}").one()
+
+            # Sanity checks.
+            assert result.keyspace_name == test_keyspace
+            assert result.type == "view"
+            assert result.name == mv_name
+
+            assert hasattr(result, "create_statement")
+            assert f"CREATE MATERIALIZED VIEW {test_keyspace}.{mv_name}" in result.create_statement
+        finally:
+            cql.execute(f"DROP INDEX {test_keyspace}.{index_name}")
```

Comment:
By the way, the try and finally is not strictly necessary in this test, because the whole thing is wrapped in a new_test_table() context manager which drops the table when the block ends - and dropping a table automatically drops its indexes (unlike materialized views).
But perhaps the try/finally makes all of this more explicit, so no need to get rid of it.